---
title: "03-methylation"
output: html_document
---

```{r}
library("tidyverse")
library(rtracklayer)
library(karyoploteR)
```


```{r}

duck.genome <- toGRanges(data.frame(chr=c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18" ), start=c(1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 ), end=c(89643857,	69596280,	57743597, 65288255, 67248332, 61759565, 43120122, 61151155, 38581958, 53961475, 51449921, 50438331, 44396874, 45393038, 47938513, 31980953, 34923512,	27737463)))


kp <- plotKaryotype(genome = duck.genome)

```


```{bash}
cat feature-tracks/EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph | sed 's/PGA_scaffold1__77_contigs__length_89643857/1/g' |  grep -v "scaffold" > feature-tracks/scf_01-meth.bedgraph

tail feature-tracks/scf_01-meth.bedgraph
```



```{r}
scf01_meth.bg <- read.csv("feature-tracks/scf_01-meth.bedgraph", sep = " ", header = FALSE)

head(scf01_meth.bg)


scf01_meth_75.bedGraph <- filter(scf01_meth.bg, V4 > 75)

head(scf01_meth_75.bedGraph)
```

```{r}
write_delim(scf01_meth_75.bedGraph, "feature-tracks/scf01_meth_75.bedGraph", delim = "\t", col_names = FALSE)
```




```{r}
scf01_meth_75 = import.bedGraph("feature-tracks/scf01_meth_75.bedGraph")

```

```{r}

kp <- plotKaryotype(genome=duck.genome, plot.type=2, chromosomes=c("1"))

kpAxis(kp, ymin=0, ymax=100, r0=NULL, r1=NULL, data.panel=1, col="gray50", cex=0.5)


kpPoints(kp, data=scf01_meth_75, x=scf01_meth_75$ranges, y=scf01_meth_75$score,
           ymin=0, ymax=100, data.panel=1, r0=NULL, r1=NULL, col=adjustcolor("pink", 0.5), pch=".", cex=0.2)

kpPlotDensity(kp, data=scf01_meth_75, window.size = 1000, data.panel=2, col=adjustcolor("green", 0.5), lwd=0.5, border=adjustcolor("green", 0.5), r0=0, r1=1) 
  
kpPlotDensity(kp, data=scf01_gene,window.size = 100000, data.panel=1)  


#kpPlotRegions(kp, scf01_meth_75, col="blue", r0=0, r1=.2, avoid.overlapping = TRUE, num.layers = 2)


```


