---
title: "06-methylation-heatmap"
output: html_document
---



```{r}
library("tidyverse")
library(rtracklayer)
library(karyoploteR)
```


```{r}

duck.genome <- toGRanges(data.frame(chr=c("Scaffold_01", "Scaffold_02", "Scaffold_03", "Scaffold_04", "Scaffold_05", "Scaffold_06", "Scaffold_07", "Scaffold_08", "Scaffold_09", "Scaffold_10", "Scaffold_11", "Scaffold_12", "Scaffold_13", "Scaffold_14", "Scaffold_15", "Scaffold_16", "Scaffold_17", "Scaffold_18" ), start=c(1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 ), end=c(89643857,	69596280,	57743597, 65288255, 67248332, 61759565, 43120122, 61151155, 38581958, 53961475, 51449921, 50438331, 44396874, 45393038, 47938513, 31980953, 34923512,	27737463)))


kp <- plotKaryotype(genome = duck.genome)

```

```{r}
#methylation data from concat of all EPI reads at minimum of 10x coverage
#500MB file
PgAll <- read_delim("http://gannet.fish.washington.edu/seashell/snaps/PgALL.bed", delim = "\t", col_names = FALSE)
names(PgAll)<-make.names(names(PgAll),unique = TRUE)
```

```{r}
head(PgAll)
```



```{r}
PgALL.meth <- (PgAll %>% mutate(name = "meth") %>% select (X1, X2, X3, name, X4))
```


```{r}
write.table(PgALL.meth, "feature-tracks/PgALLmeth.bed", col.names = FALSE, row.names = FALSE, quote = FALSE)

```

```{r}
Pg_01.meth <- (PgAll %>% mutate(name = "meth") %>% select (X1, X2, X3, name, X4) %>% filter(X1 == "Scaffold_01"))
```


```{r}
head(PgAll)
PgAll %>% filter(X1 != "Scaffold_01 ") 
```


```{r}
PgAll %>% mutate(name = "meth") %>% filter(X1 == "Scaffold_01") %>% select (X1, X2, X3, name, X4) %>% head
```

```{r}
write.table(Pg_01.meth, "feature-tracks/Pg01meth.bed", col.names = FALSE, row.names = FALSE, quote = FALSE)

```

```{bash}
head feature-tracks/PgALLmeth.bed
```




```{r}
PgALL_meth =  import("feature-tracks/PgALLmeth.bed")
```


```{r}
head(PgALL_meth)
```


```{r}
duck.genome.old <- toGRanges(data.frame(chr=c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18" ), start=c(1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 ), end=c(89643857,	69596280,	57743597, 65288255, 67248332, 61759565, 43120122, 61151155, 38581958, 53961475, 51449921, 50438331, 44396874, 45393038, 47938513, 31980953, 34923512,	27737463)))
```



```{r}
kp.old <- plotKaryotype(genome = duck.genome.old, plot.type=1)



kpPlotHorizon(kp.old, scf01_meth_75, y="3", colors = c("green", "black", "red"))


kpPlotHorizon()


```






